src/DIRAC/DataManagementSystem/scripts/dirac_dms_filecatalog_cli.py

```diff
@@ -20,50 +20,66 @@
 
   FC:/>
 """
-import sys
 
 from DIRAC.Core.Base.Script import Script
 
 
 @Script()
 def main():
-    fcType = "FileCatalog"
+    fcType = None
+    catalog = None
     Script.registerSwitch("f:", "file-catalog=", f"   Catalog client type to use (default {fcType})")
     Script.parseCommandLine(ignoreErrors=False)
 
-    from DIRAC import gConfig, exit as dexit
-    from DIRAC.Resources.Catalog.FileCatalogFactory import FileCatalogFactory
-
-    fcType = gConfig.getValue("/LocalSite/FileCatalog", "")
-
-    res = gConfig.getSections("/Resources/FileCatalogs", listOrdered=True)
-    if not res["OK"]:
-        dexit(1)
-    fcList = res["Value"]
-    if not fcType:
-        if res["OK"]:
-            fcType = res["Value"][0]
+    from DIRAC import exit as dexit
 
     for switch in Script.getUnprocessedSwitches():
         if switch[0].lower() == "f" or switch[0].lower() == "file-catalog":
             fcType = switch[1]
 
     if not fcType:
-        print("No file catalog given and defaults could not be obtained")
-        sys.exit(1)
+        # A particular catalog is not specified, try to instantiate the catalog container
+        from DIRAC.Resources.Catalog.FileCatalog import FileCatalog
+
+        catalog = FileCatalog()
+        if not catalog.valid:
+            print("Failed to create the FileCatalog container. Try to use a specific catalog with -f option")
+            dexit(-1)
+        result = catalog.getMasterCatalogNames()
+        if not result["OK"]:
+            print("Failed to get the Master catalog name for the FileCatalog container")
+            dexit(-1)
+        masterCatalog = result["Value"][0]
+        readCatalogs = [c[0] for c in catalog.getReadCatalogs()]
+        writeCatalogs = [c[0] for c in catalog.getWriteCatalogs()]
+        allCatalogs = list(set([masterCatalog] + readCatalogs + writeCatalogs))
+
+        if len(allCatalogs) == 1:
+            # If we have a single catalog in the container, let's use this catalog directly
+            fcType = allCatalogs[0]
+            catalog = None
+        else:
+            print("Starting FileCatalog container client:")
```

Comment:
Why does this say "starting", but is it actually starting anything?
And what is a "FileCatalog Container client"?
Can you add a comment here what is happening and why?

Reply:
OK, I'll try to make better comments and printouts

Reply:
Thanks! Things are clearer now!